### ISSUE TITLE
Setting statement_limit databsae config option to 0 results in crashes

### ISSUE THREAD

**Issue by nathansamson** [rails/rails#23075]
This can happen by accident when trying to read from an empty enviroment variable for example

`PG::DuplicatePstatement: ERROR: prepared statement "a1" already exists`

Ideally this should just disable the statements cache altogether (or just fallback to the normal default, although that feels less intuitive)

Happened with rails 4.2.5 on postgresql database

**Comment by renanoronfle** [comment]
can you provide example?

**Comment by sgrif** [comment]
The correct way to disable prepared statements is with `prepared_statements: false`. I'd be against making any changes to setting the statement limit to 0, as it's perfectly reasonable for a backend to work with only unnamed statements, even if that's not happening with PG in this specific case.

**Comment by haydenmellins** [comment]
Hi Nathan,,

I can help you for this and need more details. As I can see you are using postgresql database. Please msg me more details so that give you an understanding.

**Comment by nathansamson** [comment]
Hi,

I know what the problem is, and my application is currently running nicely and without any problems,

My config included
~~~
statement_limit: <%= ENV['PREPARED_STATEMENTS_LIMIT'].to_i || 50 %>
~~~
Somehow I hoped / thought that .to_i on a blank/nil string would return nil and so the || case would kick in.
This obviously did not happen, and the .to_i returned 0 resulting in the config set to 0.

And from that moment on I got weird exceptions on all page loads.

So my argument is:
It should not be possible to have such an invalid config so that the application causes crashes (obviously if you set for example an invalid hostname there is nothing rails can do than to show you "Can not connect to database: host unreachable" or something).
But for a setting like this I see 3 options:

A) Allow the value 0 and just don't use the cache but use prepared statements where possible (preferred). This might already be the case for mysql, but certainly not for postgres.
B) Interpret the value 0 as "disable prepared statements" (my original idea, but might indeed not be correct)
C) Throw a clear error (preferably on startup so the app won't even start, and heroku like inspired deploys will not cause outage at all) that the value of this particular setting is set to 0 which is not allowed. (I had this problem when I only made a change to that area, so I knew where to look, but if you get this error during a large deploy with lots of changes it might not be obvious where to look)
